refactor: replace puppeteer with astral for screenshot and close #2919

SisyphusZheng opened the pull request:
due to resolves the `NotFound: No such file or directory` error in the `www/utils/screenshot.ts` script, which was caused by improper file path construction. This error prevented screenshots from being saved correctly. close #2919 

-- update --patch1
## Current Status
- Using npm:puppeteer for webpage screenshots
- npm lifecycle script warnings present
- deno-puppeteer is no longer maintained
- npm:puppeteer still has compatibility issues

## References
- [astral](https://github.com/lino-levan/astral): Browser automation library designed for Deno
- [deno-puppeteer#92](https://github.com/lucacasonato/deno-puppeteer/issues/92): Recent discussion about deno-puppeteer

## Changes Made
1. Dependency Replacement
   - Replaced npm:puppeteer with jsr:@astral/astral
   - Removed npm lifecycle script warnings

2. Dependency Management Optimization
   - Centralized all imports in deno.json
   - Using jsr: and URL imports

## Testing Steps
1. Run screenshot command:
```bash
deno task screenshot https://google.com google
```

2. Check output:
   - Verify no npm warnings
   - Validate generated image files:
     - ./www/static/showcase/google2x.jpg
     - ./www/static/showcase/google1x.jpg
   - Confirm image quality

3. Run complete test suite:
```bash
deno task ok
```
- Ensure all tests pass

4. Verify functionality:
   - Screenshot feature works correctly
   - Image processing (2x/1x) functions properly

SisyphusZheng committed:
fix: screenshot script #2919

his PR resolves the `NotFound: No such file or directory` error in the `www/utils/screenshot.ts` script, which was caused by improper file path construction. This error prevented screenshots from being saved correctly.

iuioiua commented on www/utils/screenshot.ts:
```diff
@@ -18,13 +18,17 @@ const browser = await puppeteer.launch({
+const outputDir = new URL("../static/showcase/", import.meta.url);
+
+const jpeg2x = new URL(`${id}2x.jpg`, outputDir).pathname;
 await Deno.writeFile(jpeg2x, await image2x.encodeJPEG(80));
```
```suggestion

await Deno.writeFile(`./www/static/showcase/${id}2x.jpg`, await image2x.encodeJPEG(80));
```

iuioiua commented on www/utils/screenshot.ts:
```diff
@@ -18,13 +18,17 @@ const browser = await puppeteer.launch({
-const jpeg1x = import.meta.resolve(`../static/showcase/${id}1x.jpg`);
+const jpeg1x = new URL(`${id}1x.jpg`, outputDir).pathname;
 const image1x = image2x.resize(image2x.width / 2, Image.RESIZE_AUTO);
 await Deno.writeFile(jpeg1x, await image1x.encodeJPEG(80));
```
```suggestion
const image1x = image2x.resize(image2x.width / 2, Image.RESIZE_AUTO);
await Deno.writeFile(`./www/static/showcase/${id}1x.jpg`, await image1x.encodeJPEG(80));
```

iuioiua reviewed (changes requested):
These should work fine too.

SisyphusZheng committed:
Update www/utils/screenshot.ts

Co-authored-by: Asher Gomez <ashersaupingomez@gmail.com>

SisyphusZheng committed:
Update www/utils/screenshot.ts

Co-authored-by: Asher Gomez <ashersaupingomez@gmail.com>

iuioiua reviewed (changes requested):
Please run `deno fmt`. Also, are the `deno.lock` file changes needed?

SisyphusZheng committed:
fix: fmt script and update denolock

SisyphusZheng committed:
refactor: replace puppeteer with astral for screenshot

## Current Status
- Using npm:puppeteer for webpage screenshots
- npm lifecycle script warnings present
- deno-puppeteer is no longer maintained
- npm:puppeteer still has compatibility issues

## References
- [astral](https://github.com/lino-levan/astral): Browser automation library designed for Deno
- [deno-puppeteer#92](https://github.com/lucacasonato/deno-puppeteer/issues/92): Recent discussion about deno-puppeteer

## Changes Made
1. Dependency Replacement
   - Replaced npm:puppeteer with jsr:@astral/astral
   - Removed npm lifecycle script warnings

2. Dependency Management Optimization
   - Centralized all imports in deno.json
   - Using jsr: and URL imports

## Testing Steps
1. Run screenshot command:
```bash
deno task screenshot https://google.com google
```

2. Check output:
   - Verify no npm warnings
   - Validate generated image files:
     - ./www/static/showcase/google2x.jpg
     - ./www/static/showcase/google1x.jpg
   - Confirm image quality

3. Run complete test suite:
```bash
deno task ok
```
- Ensure all tests pass

4. Verify functionality:
   - Screenshot feature works correctly
   - Image processing (2x/1x) functions properly

iuioiua commented on www/deno.json:
These changes aren't needed right now if #2900 is merged soon.

SisyphusZheng replied:
Thx, I'll track #2900 and follow up once it's merged.👏

marvinhagemeister reviewed (approved):
LGTM, thanks!
